src/kernels/jupyter/jupyterConnection.ts

```diff
@@ -129,7 +129,7 @@ export class JupyterConnection implements IExtensionSyncActivationService {
         }
     }
 
-    private getServerUri(uri: string): IJupyterServerUri | undefined {
+    public getServerUri(uri: string): IJupyterServerUri | undefined {
```

Comment:
I think its best to use existing API.
You can use 
```
const idAndHandle = extractJupyterServerHandleAndId(uri);
if (idAndHandle) {
    const serverUri = IJupyterUriProviderRegistration.getJupyterServerUri(idAndHandle.id, idAndHandle.handle)
}
```

This way we don't have to make private stuff public.

Reply:
@DonJayamanne thanks, I'm not aware of that API, `IJupyterUriProviderRegistration.getJupyterServerUri` is better!